compiler-plugin/src/main/java/io/ballerina/lib/data/jsondata/compiler/JsondataTypeValidator.java

```diff
@@ -205,6 +215,11 @@ private boolean isSupportedUnionMemberType(TypeSymbol typeSymbol) {
         }
     }
 
+    private TypeSymbol getReferredSymbol(TypeSymbol typeSymbol) {
+        return typeSymbol.typeKind() == TypeDescKind.TYPE_REFERENCE ?
+                ((TypeReferenceTypeSymbol) typeSymbol).typeDescriptor() : typeSymbol;
```

Comment:
```suggestion
                getReferredSymbol(((TypeReferenceTypeSymbol) typeSymbol).typeDescriptor()) : typeSymbol;
```

Reply:
We are not supporting intersections yet, right? If we are supporting intersection types we can use the same logic as in [here](https://github.com/ballerina-platform/ballerina-lang/blob/06cd3a3d89b76d8cadbc9eb2aa9954fa454104fa/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java#L316).

Reply:
We are supporting the intersection. I will update the code as you suggested.

Reply:
Updated the code https://github.com/ballerina-platform/module-ballerina-data.jsondata/pull/4/commits/b2431029f62638dddb233fed00762589e866d2a0